ipn/ipnlocal/network-lock.go

```diff
@@ -430,6 +430,45 @@ func (b *LocalBackend) NetworkLockKeyTrustedForTest(keyID tkatype.KeyID) bool {
 	return b.tka.authority.KeyTrusted(keyID)
 }
 
+// NetworkLockSign signs the given node-key and submits it to the control plane.
+// rotationPublic, if specified, must be an ed25519 public key.
+func (b *LocalBackend) NetworkLockSign(nodeKey key.NodePublic, rotationPublic []byte) error {
+	b.mu.Lock()
+	if b.tka == nil {
+		b.mu.Unlock()
+		return errNetworkLockNotActive
+	}
+	if !b.tka.authority.KeyTrusted(b.nlPrivKey.KeyID()) {
+		b.mu.Unlock()
+		return errors.New("this node is not trusted by network lock")
```

Comment:
I wonder if this should be a warning rather than an error. It might be trusted in future, making the signature retroactively acceptable. Or maybe that's actually a terrible idea.

Reply:
I think I want the default to be 'only let you sign if the signature will work', but maybe we want a flag or something in future to let you submit a signature that doesnt work now but will shortly.